e2e/rbd.go

```diff
@@ -3338,6 +3340,199 @@ var _ = Describe("RBD", func() {
+							e2elog.Failf("failed to delete storageclass: %v", err)
+						}
+					}()
+					err = createRBDSnapshotClass(f)
+					if err != nil {
+						e2elog.Failf("failed to create VolumeSnapshotClass: %v", err)
+					}
+					defer func() {
+						err = deleteRBDSnapshotClass()
+						if err != nil {
+							e2elog.Failf("failed to delete VolumeSnapshotClass: %v", err)
+						}
+					}()
+					// validate filesystem mode PVC
+					err = validateBiggerPVCFromSnapshot(f,
+						pvcPath,
+						appPath,
+						snapshotPath,
+						pvcClonePath,
+						appClonePath)
+					if err != nil {
+						e2elog.Failf("failed to validate restore bigger size clone: %v", err)
+					}
+					// validate block mode PVC
+					err = validateBiggerPVCFromSnapshot(f,
+						rawPvcPath,
+						rawAppPath,
+						snapshotPath,
+						pvcBlockRestorePath,
+						appBlockRestorePath)
+					if err != nil {
+						e2elog.Failf("failed to validate restore bigger size clone: %v", err)
+					}
+				})
+
+				By("restore snapshot to bigger size encrypted PVC with VaultKMS", func() {
+					err := deleteResource(rbdExamplePath + "storageclass.yaml")
+					if err != nil {
+						e2elog.Failf("failed to delete storageclass: %v", err)
+					}
+					scOpts := map[string]string{
+						"encrypted":       "true",
+						"encryptionKMSID": "vault-test",
+					}
+					err = createRBDStorageClass(f.ClientSet, f, defaultSCName, nil, scOpts, deletePolicy)
+					if err != nil {
+						e2elog.Failf("failed to create storageclass: %v", err)
+					}
+					defer func() {
+						err = deleteResource(rbdExamplePath + "storageclass.yaml")
+						if err != nil {
+							e2elog.Failf("failed to delete storageclass: %v", err)
+						}
+					}()
+					err = createRBDSnapshotClass(f)
+					if err != nil {
+						e2elog.Failf("failed to create VolumeSnapshotClass: %v", err)
+					}
+					defer func() {
+						err = deleteRBDSnapshotClass()
+						if err != nil {
+							e2elog.Failf("failed to delete VolumeSnapshotClass: %v", err)
+						}
+					}()
+					// validate filesystem mode PVC
+					err = validateBiggerPVCFromSnapshot(f,
+						pvcPath,
+						appPath,
+						snapshotPath,
+						pvcClonePath,
+						appClonePath)
+					if err != nil {
+						e2elog.Failf("failed to validate restore bigger size clone: %v", err)
+					}
+					// validate block mode PVC
+					err = validateBiggerPVCFromSnapshot(f,
+						rawPvcPath,
+						rawAppPath,
+						snapshotPath,
+						pvcBlockRestorePath,
+						appBlockRestorePath)
+					if err != nil {
+						e2elog.Failf("failed to validate restore bigger size clone: %v", err)
+					}
+				})
+
+				By("validate image deletion", func() {
+					validateRBDImageCount(f, 0, defaultRBDPool)
+					err := waitToRemoveImagesFromTrash(f, defaultRBDPool, deployTimeout)
+					if err != nil {
+						e2elog.Failf("failed to validate rbd images in pool %s trash: %v", defaultRBDPool, err)
+					}
+				})
+			})
+
+			By("clone PVC to a bigger size PVC", func() {
```

Comment:
Instead of adding a new test can we add one more clone case to the "exisitng" clone test ?

Reply:
all clone tests need not go through the resize tests adding a separate test so that this can be reused for cephfs.